DOCS-878: Apply Matt Vella's feedback to extra params (#1468)
viamrobotics/docs: docs/program/use-extra-params.md

  - "/program/sdks/use-extra-params"
---

How to [utilize](#utilize) and [define](#define) the `extra` parameters that many {{< glossary_tooltip term_id="resource" text="resource" >}} [API methods](/program/apis/) offer in the Go and Python SDKs.

## Utilize

> You can use `extra` parameters with modular {{< glossary_tooltip term_id="resource" text="resource" >}} implementations that are new *models* of built-in resource types.

Reviewer: ```suggestion
You can use `extra` parameters with modular {{< glossary_tooltip term_id="resource" text="resource" >}} implementations that are *models* of built-in resource types.
```
Don't think we need the word 'new', as I guess any modular resource is 'new'?

For example, a new model of [sensor](/components/sensor/), or a new model of [SLAM Service](/services/slam/).

> The `extra` parameters in that built-in resource type's [API](/extend/modular-resources/#apis) give users the option to pass information to a resource's driver that isn't specified as a parameter for all models of the resource type.

Reviewer: ```suggestion
The `extra` parameters in that built-in resource type's [API](/extend/modular-resources/#apis) allow users to pass information to a resource's driver that isn't specified as a parameter for all models of the resource type.
```

> This is necessary to keep the API of resource types consistent across, for example, all models of [motor](/components/motor/) or all models of [camera](/components/camera).

Author: ```suggestion
This is necessary to keep the API of resource types consistent across, for example, all models of [motor](/components/motor/) or all models of [camera](/components/camera/).
```
